src/main/cluster.ts

```diff
@@ -138,6 +139,7 @@ export class Cluster implements ClusterModel, ClusterState {
     if (ipcMain) {
       this.eventDisposers.push(
         reaction(() => this.getState(), () => this.pushState()),
+        reaction(() => this.preferences, () => this.contextHandler.setupPrometheus(this.preferences)),
```

Comment:
Are you planning to react to any changes of `this.preference.*` internals? (or only when whole preferences object overwritten?) Otherwise probably `toJS(this.preferences, {recurseEverything: true})`?

Reply:
basically I'm interested only in `prometheus` and `prometheusProvider`. Any easy way react only changes in those two properties?

Reply:
Maybe like this?

_cluster-store.ts_
```ts
export interface ClusterPreferences extends ClusterPrometheusPreferences {
  terminalCWD?: string;
  clusterName?: string;
  iconOrder?: number;
  icon?: string;
  httpsProxy?: string;
}

export interface ClusterPrometheusPreferences {
  prometheus?: {
    namespace: string;
    service: string;
    port: number;
    prefix: string;
  };
  prometheusProvider?: {
    type: string;
  };
}
```

_cluster.ts_
```ts
  @computed get prometheusPreferences(): ClusterPrometheusPreferences {
    const { prometheus, prometheusProvider } = this.preferences;
    return toJS({ prometheus, prometheusProvider }, {
      recurseEverything: true,
    })
  }
// and just react to computed prop?
reaction(() => this.prometheusPreferences, prefs => console.log(prefs))
```

Reply:
Thanks got it working with small modifications!